packages/nuxt/src/vite/sourceMaps.ts

```diff
@@ -1,52 +1,93 @@
+      // Add Sentry plugin
       viteInlineConfig.plugins = viteInlineConfig.plugins || [];
-      viteInlineConfig.plugins.push(sentryPlugin);
-
-      const sourceMapsPreviouslyEnabled = viteInlineConfig.build?.sourcemap;
-
-      if (moduleOptions.debug && !sourceMapsPreviouslyEnabled) {
-        // eslint-disable-next-line no-console
-        console.log('[Sentry]: Enabled source maps generation in the Vite build options.');
-        if (!moduleOptions.sourceMapsUploadOptions?.sourcemaps?.filesToDeleteAfterUpload) {
-          // eslint-disable-next-line no-console
-          console.warn(
-            `[Sentry] We recommend setting the \`sourceMapsUploadOptions.sourcemaps.filesToDeleteAfterUpload\` option to clean up source maps after uploading.
-[Sentry] Otherwise, source maps might be deployed to production, depending on your configuration`,
-          );
-        }
-      }
+      viteInlineConfig.plugins.push(sentryVitePlugin(getPluginOptions(moduleOptions)));
 
+      // Enable source maps
       viteInlineConfig.build = viteInlineConfig.build || {};
       viteInlineConfig.build.sourcemap = true;
+
+      logDebugInfo(moduleOptions, viteInlineConfig.build?.sourcemap);
+    }
+  });
+
+  nuxt.hook('nitro:config', (nitroConfig: NitroConfig) => {
+    if (sourceMapsEnabled && !nitroConfig.dev) {
+      if (nitroConfig.rollupConfig) {
+        // Add Sentry plugin
+        if (!Array.isArray(nitroConfig.rollupConfig.plugins)) {
+          nitroConfig.rollupConfig.plugins = nitroConfig.rollupConfig.plugins ? [nitroConfig.rollupConfig.plugins] : [];
+        }
+
+        nitroConfig.rollupConfig.plugins.push(sentryRollupPlugin(getPluginOptions(moduleOptions)));
+
+        // Enable source maps
+        nitroConfig.rollupConfig.output = nitroConfig?.rollupConfig?.output || {};
+        nitroConfig.rollupConfig.output.sourcemap = true;
+        nitroConfig.rollupConfig.output.sourcemapExcludeSources = false; // Adding "sourcesContent" to the source map (Nitro sets this eto `true`)
+
+        logDebugInfo(moduleOptions, nitroConfig.rollupConfig.output?.sourcemap);
+      }
     }
   });
 }
+
+/**
+ * Normalizes the beginning of a path from e.g. ../../../ to ./
+ */
+function normalizePath(path: string): string {
+  return path.replace(/^(\.\.\/)+/, './');
+}
+
+function getPluginOptions(moduleOptions: SentryNuxtModuleOptions): SentryVitePluginOptions | SentryRollupPluginOptions {
+  const sourceMapsUploadOptions = moduleOptions.sourceMapsUploadOptions || {};
+
+  return {
+    org: sourceMapsUploadOptions.org ?? process.env.SENTRY_ORG,
+    project: sourceMapsUploadOptions.project ?? process.env.SENTRY_PROJECT,
+    authToken: sourceMapsUploadOptions.authToken ?? process.env.SENTRY_AUTH_TOKEN,
+    telemetry: sourceMapsUploadOptions.telemetry ?? true,
+    sourcemaps: {
+      assets: sourceMapsUploadOptions.sourcemaps?.assets ?? ['./.output/public/**/*', './.output/server/**/*'],
```

Comment:
m: I am wondering whether we will upload the same exact thing twice here because the plugin runs two times, one for nitro and one for nuxt. Is there a way we can avoid (too much) duplication? Sometimes it's unavoidable but here it might be possible to avoid it.